wurst/extensions/UnitExtensions.wurst

```diff
@@ -134,19 +134,36 @@ public function unit.setCreepGuard(bool value)
 
 public function unit.addOrStackItem(item itm) returns bool
     //Stackables
-    if isItemStackable(itm.getTypeId())
-        if not this.isItemFullyStacked(itm.getTypeId()) and this.hasItemById(itm.getTypeId())
-            this.stackItem(itm)
-            return true
-        else if not this.isInventoryFull()
-            this.addItemHandle(itm)
-            return true
+    var result = false
+    let itemTypeId = itm.getTypeId()
+    if isItemStackable(itemTypeId)
+        let itemPos = itm.getPos()
+        this.stackItem(itm)
+        if itm.isAlive()
+            //Add leftover stacks if they were left
+            var charges = itm.getCharges()
+            let maxStack = getItemMaxStacks(itemTypeId)
+            while charges > 0
+                if this.isInventoryFull()
+                    result = false
+                    break
+                if charges > maxStack
+                    let newItemStack = createItem(itemTypeId, itemPos)
+                    newItemStack.setCharges(maxStack)
+                    this.addItemHandle(newItemStack)
+                    charges -= maxStack
+                    result = true

```

Comment:
If runtime gets here, `result` would be true regardless of `charges > maxStack` check outcome.